mobile/lib/providers/folder.provider.dart

```diff
@@ -0,0 +1,22 @@
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:immich_mobile/models/folder/root_folder.model.dart';
+import 'package:immich_mobile/services/folder.service.dart';
+
+class FoldersNotifier extends StateNotifier<AsyncValue<RootFolder>> {
+  final FolderService _folderService;
+
+  FoldersNotifier(this._folderService) : super(const AsyncLoading()) {
+    fetchFolders();
+  }
+
+  Future<void> fetchFolders() async {
```

Comment:
You can use the `AsyncValue`'s methods in this function to represent the loading/error states